app/src/main/java/org/sopt/official/feature/attendance/model/MidtermAttendance.kt

```diff
@@ -0,0 +1,28 @@
+package org.sopt.official.feature.attendance.model
+
+import androidx.annotation.DrawableRes
+import org.sopt.official.R
+
+sealed class MidtermAttendance(
+    @DrawableRes val imageResId: Int,
+    val isFinished: Boolean,
+    val description: String
+) {
+    class NotYet(attendanceType: AttendanceType) : MidtermAttendance(
+        imageResId = R.drawable.ic_attendance_state_nothing,
+        isFinished = false,
+        description = attendanceType.type
+    )
+
+    class Present(attendanceAt: String) : MidtermAttendance(
+        imageResId = R.drawable.ic_attendance_state_yes,
+        isFinished = true,
+        description = attendanceAt
+    )
+
+    object Absent : MidtermAttendance(
```

Comment:
```suggestion
    data object Absent : MidtermAttendance(
```
이거 warning 안떴나요?

Reply:
아 자동으로 바뀌는거 외에 워닝으로만 뜨는것도 있나요 !?

Reply:
object 밑에 뭔가 warning 뜨는게 있지 않았나요? 1.9.0부터 object는 data object로 변경하는 것을 권장하고 있어서요.

Reply:
워닝이 뜨네요! 그런데 그 이유까지 알고싶어서 코틀린 공식 문서를 확인해봤습니다.

Data object declarations are particularly useful for sealed hierarchies like [sealed classes or sealed interfaces](https://kotlinlang.org/docs/sealed-classes.html), since they allow you to maintain symmetry with any data classes you may have defined alongside the object. In this example, declaring EndOfFile as a data object instead of a plain object means that it will get the toString() function without the need to override it manually:

요약하면 sealed 등에서 다른 요소들과 대칭성을 위해 data object를 권장한다는 말 같은데, 이는 다른 클래스들도 data class 일 경우 toString 등의 추가된 함수들을 object 또한 사용할 수 있게 한다는 의미인 것 같습니다.

그러나 현재 sealed class에서 다른 클래스들이 일반 class인 점을 생각하면 data 키워드를 붙이지 않아도 대칭성 면에서는 상관없다는 생각이 듭니다.

그러나 해당 값이 컴포저블의 파라미터로 사용되기 때문에 data class의 비교 기능이 있으면 리컴포지션을 최소화할 수 있다고 판단하여(파라미터로 입력된 값이 변경되면 해당 컴포저블을 다시 그리는 것으로 아는데 맞나요??) 다른 파라미터로 사용되는 클래스들도 모두 data class로 변경하려고 하는데 맞을까요?

Reply:
> 요약하면 sealed 등에서 다른 요소들과 대칭성을 위해 data object를 권장한다는 말 같은데, 이는 다른 클래스들도 data class 일 경우 toString 등의 추가된 함수들을 object 또한 사용할 수 있게 한다는 의미인 것 같습니다.

맞습니다.

> 그러나 현재 sealed class에서 다른 클래스들이 일반 class인 점을 생각하면 data 키워드를 붙이지 않아도 대칭성 면에서는 상관없다는 생각이 듭니다.

이게 sealed를 상속을 받든 안받든 경고문은 띄워질거에요. 현재 코틀린에선 모든 object에 대해 data object로 전환할 것을 권고합니다. 이는 리컴포지션과 대칭성 등과 관련이 없습니다.

Reply:
<img width="152" alt="스크린샷 2024-05-31 오후 1 17 30" src="https://github.com/sopt-makers/sopt-android/assets/108331578/45d5298a-4e8c-4471-8f20-17ac82cae677">

음 저는 sealed 상속을 안 받으면 경고문이 안 나오는 것 같은데, 혹시 data object로 전환할 것을 권고하는 링크 또는 이유가 있나요??

Reply:
음 그렇군요..? 일단 data object로 권고하는건 object 로그를 남기면 알 수 있을텐데, 얘가 로그남기면 주소가 떠서 내용을 못봅니다.하지만 data object로 만들면 내용은 볼 수 있어서 그 부분때매 권고하는거에요

Reply:
아하 data object로 만들면 toString()에서 오브젝트명만 나오므로 내용을 볼 수 있게 되겠군요. 새로운 내용 알아갑니다 감사합니다!!